library/sha256.c

```diff
@@ -22,8 +22,17 @@
  *  http://csrc.nist.gov/publications/fips/fips180-2/fips180-2.pdf
  */
 
-#if defined(__aarch64__) && !defined(__ARM_FEATURE_CRYPTO) && \
-    defined(__clang__) && __clang_major__ >= 4
+#if defined(__clang__) &&  (__clang_major__ >= 4)
+
+/* Ideally, we would simply use MBEDTLS_ARCH_IS_ARMV8_A in the following #if,
+ * but that is defined by build_info.h, and we need this block to happen first. */
+#if defined(__ARM_ARCH) && (__ARM_ARCH_PROFILE == 'A')
+#if __ARM_ARCH >= 8
+#define MBEDTLS_SHA256_ARCH_IS_ARMV8_A
+#endif
+#endif
```

Comment:
Comment for #7078 which introduce a separately  macros and module. 

This is duplicate with definition in build_info. The information is put in `build_info.h` and the fix MUST be put in first line of file. How about move those definition into separate file and include the file here and in `build_info.h`.

Reply:
I considered this but it seemed excessive for one use-case. But if you are repeating this pattern in 7078, then I agree that it would make sense to separate this out.